integrations/drip/drip_test.go

```diff
@@ -0,0 +1,177 @@
+}
+
+func TestNotEnabledWhenMissingCredentials(t *testing.T) {
+	os.Setenv("DRIP_API_TOKEN", "")
+	os.Setenv("DRIP_ACCOUNT_ID", "")
+	drip := Drip{}
+	if drip.Enabled() {
+		t.Error("Should not be enabled when missing accoundId and apiToken")
+	}
+	os.Setenv("DRIP_API_TOKEN", "123")
+	if drip.Enabled() {
+		t.Error("Should not be enabled when missing accoundId")
+	}
+	os.Setenv("DRIP_API_TOKEN", "")
+	os.Setenv("DRIP_ACCOUNT_ID", "123")
+	if drip.Enabled() {
+		t.Error("Should not be enabled when missing apiToken")
+	}
+}
+
+func TestEnabledWhenCredentialsPresent(t *testing.T) {
+	os.Setenv("DRIP_API_TOKEN", "123")
+	os.Setenv("DRIP_ACCOUNT_ID", "123")
+	drip := Drip{}
+	if !drip.Enabled() {
+		t.Error("Drip should be enabled when credentials are set")
+	}
+}
+
+func TestIdentify(t *testing.T) {
+	os.Setenv("DRIP_API_TOKEN", "123")
+	os.Setenv("DRIP_ACCOUNT_ID", "321")
+	api := &APIMock{Url: "http://www.example.com"}
+	drip := Drip{}
+	drip.api = api
+	identification := integrations.Identification{
+		UserID: "123",
+		UserTraits: map[string]interface{}{
+			"email": "john@example.com",
+		},
+	}
+	err := drip.Identify(identification)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if api.Method != "POST" {
+		t.Errorf("Expected method to be POST, was: %v", api.Method)
+	}
+
+	if api.Endpoint != "subscribers" {
+		t.Errorf("Expected endpoint to be subscribers, was: %v", api.Endpoint)
+	}
+
+	expectedPayload := `{"subscribers":[{"custom_fields":{"email":"john@example.com"},"email":"john@example.com","user_id":"123"}]}`
+	if string(api.Payload) != expectedPayload {
+		t.Errorf("Expected payload: "+expectedPayload+" got: %v", api.Payload)
+	}
+}
+
+func TestTrackErrorWhenNoEmailPresent(t *testing.T) {
+	os.Setenv("DRIP_API_TOKEN", "123")
+	os.Setenv("DRIP_ACCOUNT_ID", "321")
+	api := &APIMock{Url: "http://www.example.com"}
+	drip := Drip{}
+	drip.api = api
+	event := integrations.Event{
+		Name:       "account.created",
+		UserID:     "123",
+		Properties: map[string]interface{}{},
+		Timestamp:  1234567,
+		ReceivedAt: 65,
+	}
+	err := drip.Track(event)
+	if err == nil {
+		t.Error("Expected error when no email given")
+	}
+}
+
+func TestTrack(t *testing.T) {
+	os.Setenv("DRIP_API_TOKEN", "123")
+	os.Setenv("DRIP_ACCOUNT_ID", "321")
+	api := &APIMock{Url: "http://www.example.com"}
+	drip := Drip{}
+	drip.api = api
+	event := integrations.Event{
+		Name:   "account.created",
+		UserID: "123",
+		Properties: map[string]interface{}{
+			"email": "john@example.com",
+		},
+		Timestamp:  1234567,
+		ReceivedAt: 65,
+	}
+
+	err := drip.Track(event)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if api.Method != "POST" {
+		t.Errorf("Expected method to be POST, was: %v", api.Method)
+	}
+
+	if api.Endpoint != "events" {
+		t.Errorf("Expected endpoint to be events, was: %v", api.Endpoint)
+	}
+
+	expectedData := &MockEvents{
+		Events: []MockEvent{
+			{
+				Action:     "account.created",
+				Email:      "john@example.com",
+				OccurredAt: time.Unix(1234567, 0),
```

Comment:
As it ends up being marshalled, could it be a string at this level so we would make sure that this is marshalled the way we want?